FIX address sender mailer config

Timothee-iSi opened the pull request:
the address sender field in the mailer config in the backend was ignored
the order of affectation have been changed to take into account the custom
address sender according to the sales channel.

### 1. Why is this change necessary?
The address sender in the mailer settings in the backend was ignored.
When someone wanted to custom the sender email address for the customer it was simply ignored.

### 2. What does this change do, exactly?
The only thing that change is the variable affectation order.
In first position there is the custom sender email address, and then come the default one and in last the basic information.
So if someone fill the field sender address it will be set first.

### 3. Describe each step to reproduce the issue or behaviour.
First go to mailer settings
in backend : settings -> system -> mailer
here fill the fields to configure the mail server (chose SMTP server)
set also the sender address to what you want (must be email address)
then do to any mail template
in backend : settings -> shop -> email template
here chose any of them and send a test email
receiving the mail you can see the sender email address
after correction it would be the one set in the backend and before correction it would be the actual email address

### 4. Please link to the relevant issues (if any).


### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

Crease29 commented on engine/Shopware/Bundle/PluginInstallerBundle/Service/PluginExtractor.php:
```diff
@@ -73,6 +102,7 @@ public function extract($archive)
         }
 
         $prefix = $this->getPluginPrefix($archive);
+        $this->validatePluginRequirements($prefix, $archive);
```
Wouldn't it be useful to validate the plugin zip first and validate the plugin requirements after that?

shyim replied:
Upss

oktupol commented on engine/Shopware/Bundle/PluginInstallerBundle/Service/PluginExtractor.php:
```diff
@@ -213,4 +243,19 @@ private function createBackupFile($oldFile)
+        if ($xml = $archive->getFromName($prefix . '/plugin.xml')) {
+            $tmpFile = $this->downloadDir . '/' . uniqid() . '.xml';
+            file_put_contents($tmpFile, $xml);
+            $this->requirementsValidator->validate($tmpFile, $this->release->getVersion());
```
If `validate` throws an Exception, which it does, the `$tmpFile` stays in the file system until it's manually deleted. I suggest

```php
try {
    $this->requirementsValidator->validate($tmpFile, $this->release->getVersion();
} finally {
    unlink($tmpFile);
}
```

shyim replied:
You are right, i will update it later. Thanks ❤️

oktupol commented on engine/Shopware/Bundle/PluginInstallerBundle/Service/PluginExtractor.php:
```diff
@@ -254,7 +254,11 @@ private function validatePluginRequirements($prefix, \ZipArchive $archive)
+            } finally {
+                unlink($tmpFile);
+            }
             unlink($tmpFile);
```
Now this one is obsolete :smile:

shyim replied:
Mähh, i guess finally is only called when a exception is thrown. Its removed thanks :D

tpierschel commented:
@shopwareBot  please merge this PR very soon. Our clients ran exactly in this trap. A big thanks to @shyim for fixing this.

bneumann97 commented:
Hey @shyim,

Thanks for your contribution 💙 
Could you please squash your commits?

Timothee-iSi committed:
FIX address sender mailer config

the address sender field in the mailer config in the backend was ignored
the order of affectation have been changed to take into account the custom
addresws sender according to the sales channel.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/shopware/platform?pullRequest=1549) <br/>All committers have signed the CLA.

bneumann97 commented:
The tests are failing and you forgot the checklist in the contribution template. Please always make sure to work through the checklist. The changelog file is also important.

JoshuaBehrens commented:
Is this also solving https://github.com/shopware/platform/pull/684 ?

Timothee-iSi commented:
> Is this also solving #684 ?

@JoshuaBehrens I do not think so, cause I did not have this issue

Timothee-iSi committed:
FIX address sender mailer config

creating the changelog for the issue

mitelg commented on changelog/_unreleased/2020-12-03-Fix-custom-sender-email.md:
```diff
@@ -0,0 +1,8 @@
+author_email:   timothee.motnias@epita.fr
+---
+# Core
+    * Cahnged the MailService class by changing the senderEmail variable affectation begining with the custom email address.
```
```suggestion
* Changed the `senderEmail` variable affectation beginning with the custom email address in the `Shopware\Core\Content\MailTemplate\Service\MailService`
```

Timothee-iSi committed:
Update changelog/_unreleased/2020-12-03-Fix-custom-sender-email.md

Co-authored-by: Michael Telgmann <mitelg@users.noreply.github.com>

bneumann97 commented:
@Timothee-iSi Please have a look at the failing Unit-Tests

philipreinken commented:
Hello, we're closing this pull request due to inactivity. If this change is still important to you, feel free to create a new pull request.

For more information about our contribution guidelines, see https://docs.shopware.com/en/shopware-platform-dev-en/contribution/contribution-guideline.
